File: test/extensions/arguments/with_type_info_code/cpp/JsonParamCastTest.cpp
#include "gtest/gtest.h"
#include "json_param_cast/Holder.h"
#include "zserio/DebugStringUtil.h"
#include "zserio/JsonWriter.h"

namespace json_param_cast
{

using allocator_type = Holder::allocator_type;
using string_type = zserio::string<allocator_type>;

const string_type JSON_NAME_PARAM_CAST = "json_param_cast.json";

TEST(JsonParamCastTest, toJsonFile)
{
    Holder holder(10, Int8Param(1), Int16Param(2), Int32Param(3), Int64Param(4), Float16Param(5),
            Float32Param(6), Float64Param(7), Int32Param(8));
    holder.initializeChildren();
    zserio::toJsonFile(holder, JSON_NAME_PARAM_CAST);

    Holder readHolder = zserio::fromJsonFile<Holder>(JSON_NAME_PARAM_CAST, allocator_type());
    readHolder.initializeChildren();
    ASSERT_EQ(holder, readHolder);
}

} // namespace json_param_cast

File: test/extensions/arguments/with_type_info_code/python/JsonParamCastTest.py
import os
import unittest
import zserio

from testutils import getApiDir
from testutils import getZserioApi


class JsonParamCastTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "json_param_cast.zs", extraArgs=["-withTypeInfoCode"])

    def testToJsonFile(self):
        holder = self.api.Holder(
            10,
            self.api.Int8Param(0, 1),
            self.api.Int16Param(0o30, 2),
            self.api.Int32Param(0x30, 3),
            self.api.Int64Param(0b111, 4),
            self.api.Float16Param(0.2, 5),
            self.api.Float32Param(0.2, 6),
            self.api.Float64Param(0.1, 7),
            self.api.Int32Param(10, 8),
        )
        zserio.to_json_file(holder, self.JSON_NAME_PARAM_CAST)

        readHolder = zserio.from_json_file(self.api.Holder, self.JSON_NAME_PARAM_CAST)
        self.assertEqual(holder, readHolder)

    JSON_NAME_PARAM_CAST = os.path.join(getApiDir(os.path.dirname(__file__)), "json_param_cast.json")
